application-admintools-default/src/main/java/com/xwiki/admintools/internal/uploadJob/UploadJobFileProcessor.java

```diff
@@ -0,0 +1,179 @@
+            createBackupFile(targetFile.getAbsolutePath() + ".bak", targetFile, uploadPackageJobResource, fileName,
+                status);
+        } else {
+            String extension = filePath.substring(filePath.lastIndexOf("."));
+            if (extension.equals(".jar")) {
+                String simpleFileName = extractNameBeforeVersion(fileName);
+                String directoryPath = targetFile.getParent();
+                String[] filesList = new File(directoryPath).list(
+                    ((dir, name) -> name.startsWith(simpleFileName) && name.endsWith(extension)));
+                if (filesList == null || filesList.length > 1) {
+                    JobResult log =
+                        new JobResult("adminTools.jobs.upload.assessoriginal.fail", JobResultLevel.ERROR, fileName);
+                    status.addLog(log);
+                    throw new RuntimeException(
+                        String.format("Unable to correctly assess the original file for %s.", fileName));
+                } else if (filesList.length == 1) {
+                    String oldFilePath = String.format(NEW_FILE_PATH_FORMAT, directoryPath, filesList[0]);
+                    targetFile = new File(oldFilePath);
+                    createBackupFile(String.format("%s.bak", oldFilePath), targetFile, uploadPackageJobResource,
+                        fileName, status);
+                }
+            }
+        }
+        uploadPackageJobResource.setTargetFile(targetFile);
+        return uploadPackageJobResource;
+    }
+
+    /**
+     * Get the content of an {@link XWikiAttachment} as an {@link InputStream}.
+     *
+     * @param fileRef the reference of the attachment.
+     * @return the {@link InputStream} of the attachment.
+     * @throws XWikiException if any errors occur during the retrieval of the attachment.
+     */
+    public InputStream getArchiveInputStream(String fileRef) throws XWikiException
+    {
+        XWikiContext xcontext = this.xcontextProvider.get();
+        AttachmentReference attachmentReference = attachmentReferenceResolver.resolve(fileRef);
+        XWikiDocument parentDoc = xcontext.getWiki().getDocument(attachmentReference.getDocumentReference(), xcontext);
+        return parentDoc.getAttachment(attachmentReference.getName()).getContentInputStream(xcontext);
+    }
+
+    /**
+     * Process and store the content of a {@link ZipInputStream}.
+     *
+     * @param zis the zip content to be processed.
+     * @param jobResource the resource used for storing the result.
+     * @throws IOException if any errors occur during the reading of the content.
+     */
+    public void processFileContent(ZipInputStream zis, UploadPackageJobResource jobResource) throws IOException
+    {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+        int len;
+        while ((len = zis.read(buffer)) > 0) {
+            baos.write(buffer, 0, len);
```

Comment:
This is not a good idea. You seem to be storing the content of the files in memory. This can lead to OutOfMemory quite easily. I dont think you need to store the content of the file right now. You can simply open it and process the content when you need it.